python/sparkdl/param/image_params.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Some parts are copied from pyspark.ml.param.shared and some are complementary
+to pyspark.ml.param. The copy is due to some useful pyspark fns/classes being
+private APIs.
+"""
+
+from pyspark.ml.param import Param, Params, TypeConverters
+from pyspark.sql.functions import udf
+
+from sparkdl.image.imageIO import imageArrayToStruct, imageSchema
+from sparkdl.param import SparkDLTypeConverters
+
+OUTPUT_MODES = ["vector", "image"]
+
+class HasTensorOutput(Params):
+
+    outputMode = Param(Params._dummy(), "outputMode",
+                       "How the output column should be formatted. 'vector' for a 1-d MLlib " +
+                       "Vector of floats. 'image' to format the output to work with the image " +
+                       "tools in this package.",
+                       typeConverter=SparkDLTypeConverters.supportedNameConverter(OUTPUT_MODES))    
+
+    def __init__(self):
+        super(HasTensorOutput, self).__init__()
+
+    def setOutputMode(self, value):
+        return self._set(outputMode=value)
+
+    def getOutputMode(self):
+        return self.getOrDefault(self.outputMode)
+
+
+class HasLabelCol(Params):
+
+    labelCol = Param(Params._dummy(), "labelCol",
+                     "name of the column storing the training data labels",
+                     typeConverter=TypeConverters.toString)
+    
+    labelCardinality = Param(Params._dummy(), "labelCardinality",
```

Comment:
The `label` column is expected to be numerical. The training data might not have all the label presented, so we also opt for users to provide a `cardinarlity` parameter. Most Keras models expect one-hot encoded labels. If the input labels are not in that format, we convert it.